tdesign/src/Input/styles.tsx

```diff
@@ -29,4 +33,8 @@ export const styles = {
       borderColor: "on.error.main",
     },
   },
+  ".MuiButtonBase-root": { marginRight: buttonMarginRight },
+
+  //Limit button hieght when within the input field
+  ".MuiButton-root": { maxHeight: `calc(100% - (2 * ${buttonMarginRight}))` },
```

Comment:
These styles seem a bit misplaced to me. The input defines how the button should look. In my view, such styles should be placed on the button that is injected into the input, not on the input itself (via CSS cascading).

Is there a way to move them to the injected button?

Reply:
It can be moved, but my only issue with that is that we might lose some sense of standardization. With the current placement of the styles, every button inside of a textinput would have the same `4px` (`buttonMarginRight`) margin around the sides, but that may not be true iff the styles were placed in the inject button.

Would love to hear your thoughts. I think these are the barebones styles that apply to all buttons inside a textfield, anything in the injected button `sx` would be a slight customization

Reply:
Sure, that's a valid concern. I thought about creating a new specialized component for an input with a button as an adornment and styling the adornment button in that special component, but then I thought that it may be an overkill.

A solution that may be worth it is creating a component that would wrap the adornment button. It would enforce the styles that are currently hardcoded here, so it would look like:

```ts
const InputAdornmentButtonContainer = styled("div")({
  marginRight: buttonMarginRight,
  maxHeight: `calc(100% - (2 * ${buttonMarginRight}))`,
});
```

and would be used like:

```tsx
<Input
  endAdornment={
    <InputAdornmentButtonContainer>
      <InFieldButton>...</InFieldButton>
    </InputAdornmentButtonContainer>
  }
/>
```

but that could be an overkill too :smile: 

All in all, I believe your solution to be the most well-balanced. Thanks for talking me out of my suggestion and pushing back :smile: I will resolve the thread since there is nothing actionable here anymore